lib/icinga/pluginutility.cpp

```diff
@@ -217,3 +222,50 @@ String PluginUtility::FormatPerfdata(const Array::Ptr& perfdata)
 
 	return result.str();
 }
+
+Value PluginUtility::LookupCommandPath(const String& executable)
+{
+	Value pluginPath = IcingaApplication::GetPluginPath();
+	if (pluginPath.IsEmpty() || !pluginPath.IsObjectType<Array>()) {
+		Log(LogDebug, "PluginUtility", "Can not lookup command path, PluginPath is empty or not an Array!");
+		return Empty;
+	}
+
+	Array::Ptr pathList = pluginPath;
+	ObjectLock olock(pathList);
+	String script;
+	for (const String& path : pathList) {
+		if (!Utility::PathExecutable(path))
+			continue;
+		script = Utility::ConcatPath({ path, executable });
+		Log(LogDebug, "PluginUtility", "Looking for plugin at: " + script);
+		if (Utility::PathExecutable(script)) {
+			Log(LogDebug, "PluginUtility", "Executable plugin found at: " + script);
+			return script;
```

Comment:
Please add more new lines between statements above. E.g. after `continue;`, before `if (...`.

Reply:
More importantly using '+' for log entries is bad. Use << instead. E.g. `Log(LogDebug, "PluginUtility") << "abc" << script`.